snippets/base/templates/base/includes/snippet_js.html

```diff
@@ -355,19 +358,45 @@
         }
     }
 
-    // Download the user's country using the geolocation service.
-    function downloadUserCountry() {
-        $script(GEO_URL, 'geo');
-        $script.ready('geo', function() {
-            try {
-                gSnippetsMap.set('geoCountry', geoip_country_code());
-                gSnippetsMap.set('geoLastUpdated', new Date());
-            } catch (e) {
-                // Most likely failed to load JS file. Continue on without us,
-                // we'll try again next time.
-            }
-        });
-    }
+    {% if client.channel == 'release' %}
+      function downloadUserCountry() {
+          var GEO_URL = "{{ settings.OLD_GEO_URL }}";
+          $script(GEO_URL, 'geo');
+          $script.ready('geo', function() {
+              try {
+                  gSnippetsMap.set('geoCountry', geoip_country_code());
+                  gSnippetsMap.set('geoLastUpdated', new Date());
+              } catch (e) {
+                  // Most likely failed to load JS file. Continue on without us,
+                  // we'll try again next time.
+              }
+          });
+      }
+    {% else %}
+      // Download the user's country using the geolocation service.
+      // Please do not directly use this code or Snippets key.
+      // Contact MLS team for your own credentials.
+      // https://location.services.mozilla.com/contact
+      function downloadUserCountry() {
+          var GEO_URL = "{{ settings.GEO_URL }}";
+          var request = new XMLHttpRequest();
+          request.onreadystatechange = function() {
+              if (request.readyState == 4 && request.status == 200) {
+                  var country_data = JSON.parse(request.responseText);
+
+                  try {
+                      gSnippetsMap.set('geoCountry', country_data.country_code);
+                      gSnippetsMap.set('geoLastUpdated', new Date());
+                  } catch (e) {
+                      // Most likely failed to load Data file. Continue on without us,
+                      // we'll try again next time.
+                  }
+              }
+          };
+          request.open('GET', GEO_URL, true);
+          request.send();
```

Comment:
@glogiotatidis ok, then ignore my suggestion for now. I'll look into the additional header in more detail next year. We won't do any new non-emergency deployments to the service side code this year. Since your version works now, let's go with that so you can switch this on whenever you want to :)